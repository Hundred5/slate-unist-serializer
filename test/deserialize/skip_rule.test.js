import u from "unist-builder";
import Serializer from "../../src";

const rules = [
  {
    deserialize(node, next) {
      if (node.type === "paragraph") {
        return {
          object: "block",
          type: "paragraph",
          nodes: next(node.children)
        };
      }
    }
  },
  {
    deserialize(node, next) {
      if (node.type === "root") {
        return {
          object: "document",
          nodes: node.children && next(node.children)
        };
      }
    }
  }
];

const input = u("root");

test("skipping deserialize rules", () => {
  const serializer = new Serializer({ rules });
  const output = serializer.deserialize(input).toJSON();
  expect(output).toMatchSnapshot();
});
