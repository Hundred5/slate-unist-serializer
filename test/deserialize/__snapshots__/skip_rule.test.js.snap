// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skipping deserialize rules 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [],
    "object": "document",
  },
  "object": "value",
}
`;
